src/main.cpp

```diff
@@ -3877,14 +3892,21 @@ bool static LoadBlockIndexDB()
                     } else {
                         pindex->nChainSproutValue = boost::none;
                     }
+                    if (pindex->pprev->nChainSaplingValue) {
+                        pindex->nChainSaplingValue = *pindex->pprev->nChainSaplingValue + pindex->nSaplingValue;
+                    } else {
+                        pindex->nChainSaplingValue = boost::none;
+                    }
```

Comment:
The 10 lines above this are the same as in ```bool ReceivedBlockTransactions(...)```. Is it worth it to extract a function?

Reply:
Maybe, but in a future PR.